src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/MicrosoftApiDesignGuidelinesAnalyzersResources.resx

```diff
@@ -1007,4 +1007,13 @@
     <value>{0} should define operator(s) '{1}' and Equals since it implements IComparable.</value>
     <comment>1 is a comma-separated list</comment>
   </data>
+  <data name="DoNotCatchGeneralExceptionTypesDescription" xml:space="preserve">
+    <value>General exceptions should not be caught.</value>
+  </data>
+  <data name="DoNotCatchGeneralExceptionTypesMessage" xml:space="preserve">
+    <value>A general exception such as System.Exception or System.SystemException is caught in a catch statement, or a general catch clause such as catch() is used.</value>
```

Comment:
>A general exception such as System.Exception or System.SystemException is caught in a catch statement, or a general catch clause such as catch() is used. [](start = 11, length = 153)

Can we retain the original message used in prior FXCop implementation: `Modify '{0}' to catch a more specific exception type than '{1}', or rethrow the exception`? That seems more prescriptive on how to fix the issue.

Reply:
Note your current string seems suitable to prepend to the description above. `A general exception such as System.Exception or System.SystemException is caught in a catch statement, or a general catch clause such as catch() is used. General exceptions should not be caught.`

---
In reply to: [251623549](https://github.com/dotnet/roslyn-analyzers/pull/2082#discussion_r251623549) [](ancestors = 251623549)

Reply:
I made it `Modify '{0}' to catch a more specific exception type, or rethrow the exception`, since the type of exception being caught is not particularly useful, and in the case of a generic `catch`, there's no type to specify.